### PR TITLE
Now disposes of user on disconnect

### LINKED ISSUE #223: User is not removed from user list on disconnect
Steps to reproduces:
* Open and log in with two tabs
* Note that your avatar shows up twice in user list
* Close one tab
* Note that there is still two avatars, while we'd expect one only

Note that this works on sign out (and probably ban/remove user as well). It simply doesn't seem like the function is being called on disconnect (See SocketServer.ts in backend, probably "destroyUser" function isn't being called properly on disconnect

### LINKED ISSUE #226: Your own cursor is not removed
To replicate:
Login -> create a file -> place the cursor somewhere -> press F5
Now "your own" cursor from before update remains, think we need to remove from server upon socket disconnect.

### PR DESCRIPTION
**Pull request by rehnarama**
Fixes #223 and fixes #226  (they were caused by same bug)

### CODE REVIEW

File: packages/backend/src/functions/socketServer/SocketServer.ts [hunk 1]
```diff
@@ -144,9 +145,9 @@ export default class SocketServer {
     return this.userMap.get(socketId)?.getUserData();
   }
 
-  public destroyUser(login: string): void;
-  public destroyUser(user: User): void;
-  public destroyUser(user: User | string): void {
+  public destroyUser(login: string, clearSession: boolean): void;
+  public destroyUser(user: User, clearSession: boolean): void;
+  public destroyUser(user: User | string, clearSession = true): void {
```
Comment (jowe3751) [inline packages/backend/src/functions/socketServer/SocketServer.ts]: Cant find any occurrence when this is called with a string, is this old stuff?.. The middle function declaration should suffice?

Comment (rehnarama) [reply packages/backend/src/functions/socketServer/SocketServer.ts]: Both of what you say is true!

Yes, the string argument is never used and yes, this is old stuff

File: packages/backend/src/functions/socketServer/SocketServer.ts [hunk 2]
```diff
@@ -144,9 +145,9 @@ export default class SocketServer {
     return this.userMap.get(socketId)?.getUserData();
   }
 
-  public destroyUser(login: string): void;
-  public destroyUser(user: User): void;
-  public destroyUser(user: User | string): void {
+  public destroyUser(login: string, clearSession: boolean): void;
```
Comment (jowe3751) [inline packages/backend/src/functions/socketServer/SocketServer.ts]:
```suggestion
```

File: packages/backend/src/functions/socketServer/SocketServer.ts (hunk 1 above)
Comment (jowe3751) [inline packages/backend/src/functions/socketServer/SocketServer.ts]:
```suggestion
```

File: packages/backend/src/functions/socketServer/SocketServer.ts [hunk 3]
```diff
@@ -144,9 +145,9 @@ export default class SocketServer {
...
+  public destroyUser(user: User, clearSession: boolean): void;
```
Comment (jowe3751) [inline packages/backend/src/functions/socketServer/SocketServer.ts]:
```suggestion
  public destroyUser(user: User, clearSession: boolean): void {
```

**Pull request merged**